### DISCUSSION TITLE
antd的定位是什么？pc only?

### DISCUSSION THREAD

**Discussion by shifenhutu** [ant-design/ant-design#41003] [General]
在我看来，antd很优秀，特别是完善的中文文档 

最近我需要做一个响应式的页面， 需要支持 电脑、手机

我发现 Button 组件，在手机上的行为有些不完美，排查后发现是 手机上的浏览器不支持 `user-select ` 于是我提了个 pr  https://github.com/ant-design/ant-design/pull/40994

在pr里  @MadCcc  说 `移动端不在我们的默认支持范围内` 

随后我查看了ant官网，发现 https://ant.design/docs/react/introduce-cn 确实有关于antd的定位说明
![image](https://user-images.githubusercontent.com/104620424/222144129-4ca84e5a-9c2f-45c3-9c5d-68348a54c05f.png)

想问问，antd未来是否会考虑扩大 支持范围，比如  支持 ios系统上的 safari、chrome浏览器 

> 目前 隔壁 [mui](https://mui.com/components/buttons/) 的 Button 的 `user-select `  在手机端是完美工作的 

-------

虽然我是个菜鸟，但我感觉antd有些闭门造车，如果定位模糊，，antd的贡献者会越来越累、越来越少，antd用户也会流失

![image](https://user-images.githubusercontent.com/104620424/222146622-2dfbbb28-e098-465a-9e37-2f61f1c019a7.png)
> mui 改版前，下载量远远不如antd，改版后，下载量是antd的两倍，  github start也已经超过了antd 

我是真心希望antd越来越好，但目前我看到了傲慢，固执， 或许是我一厢情愿

**Comment by MadCcc** [comment]
感谢您对 antd 的支持。

看了一下 v4 的 `user-select` 是有 autoprefix 的，v5 由于换了 cssinjs，底层能力还不完善，这部分还没支持上。

一般来说我们不会在编写样式时都记得添加厂商前缀，而是通过底层插件解决，这是 https://github.com/ant-design/ant-design/pull/40994 这个 PR 关闭的原因，欢迎留一个 issue 下来。

最后如果急需使用的话，antd 的 cssinjs 提供了 transformer 的能力，厂商前缀的问题也可以很方便地通过 transformer 解决。我相信这不是一个很复杂的方案，在 cssinjs 仓库中就有类似的例子。这是一个当前的 workaround，后续我们会完善 cssinjs 底层的能力以解决厂商前缀的问题。

**Reply by MadCcc** [reply]
https://github.com/ant-design/ant-design/issues/41004
Trace here.

**Reply by zombieJ** [reply]
MUI 默认是移动端优先，着重点的确是不同。但是这些 autoprefix 感觉的确是可以做

**Reply by shifenhutu** [reply]
谢谢回复，期待早点支持

**Comment by afc163** [comment]
1. MUI 的 npm 下载量一直都是远超 antd 的，大概是 2 到 2.5 倍之间。上图这个趋势是因为 MUI 总是在换新包名。
2. MUI Button 之所以有 prefix 是利用了 [jss-plugin-vendor-prefixer](https://www.npmjs.com/package/jss-plugin-vendor-prefixer) 的能力，可以看看 https://mui.com/material-ui/getting-started/supported-platforms/#css-prefixing 。antd 也可以提供类似的能力，就是 @MadCcc 提的自定义 transformer，是可以官方直接提供一个的，也可以由社区贡献一个到官方仓库里。当然同样要通过 StyleProvider 引入使用。

**Reply by shifenhutu** [reply]
谢谢回复，期待antd越来越好， 建议时间条件允许的情况下，可以多写一些资料，方便更多人来做贡献

**Comment by afc163** [comment]
https://ant.design/docs/react/compatible-style-cn @MadCcc 这里可以补充一下 vendor-prefix 的说明
